lib/document.js

```diff
@@ -3406,20 +3396,22 @@ Document.prototype.$toObject = function(options, json) {
   // `clone` is necessary here because `utils.options` directly modifies the second input.
   defaultOptions = utils.options(defaultOptions, clone(baseOptions));
   defaultOptions = utils.options(defaultOptions, clone(schemaOptions[path] || {}));
-
   // If options do not exist or is not an object, set it to empty object
   options = utils.isPOJO(options) ? clone(options) : {};
   options._calledWithOptions = options._calledWithOptions || clone(options);
-
   let _minimize;
+
+  // console.log('Schema Options', schemaOptions.minimize)
+  // console.log('options', options)
   if (options._calledWithOptions.minimize != null) {
     _minimize = options.minimize;
   } else if (defaultOptions.minimize != null) {
     _minimize = defaultOptions.minimize;
+  } else if (options._parentOptions != null && options.getters != null) {
```

Comment:
Why `options.getters != null` here?

Reply:
it won't work without it. gh-9405 has tests already and this was the only combination that could get them all to pass.